ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -139,6 +149,14 @@ private PubsubMessageToTableRow(ValueProvider<String> tableSpecTemplate,
 
     final TableDestination tableDestination = new TableDestination(tableSpec, null,
         TIME_PARTITIONING);
+
+    if (tableRowFormat != null && tableRowFormat.get() == TableRowFormat.raw) {
+      TableRow tableRow = new TableRow();
+      AddMetadata.attributesToMetadataPayload(message.getAttributeMap()).forEach(tableRow::set);
+      tableRow.set("payload", message.getPayload());
+      return KV.of(tableDestination, tableRow);
```

Comment:
The early return here makes me think we should factor out the rest of this method to a separate `processDecoded` method or the like. I think at least this should move down below the "Resolved destination table does not exist" check.

Reply:
yeah, i agree. i just didn't like the size of diff in created 😭

Reply:
fixed